hello.cpp

```diff
@@ -0,0 +1,46 @@
+#include <thread>
+#include <boost/asio.hpp>
+
+#include "packet.h"
+
+using boost::asio::ip::tcp;
+
+const int port = 3000;
+
+void session(tcp::socket socket, const std::shared_ptr<KeyValueStore>& k) {
+  boost::system::error_code ec;
+  try {
+    for (;;) {
+      char data[Packet::HEADER_LENGTH] = { 0 };
+      size_t bytes_read = socket.read_some(boost::asio::buffer(data), ec);
+      if (ec == boost::asio::error::eof) {
+        return;
+      } else if (ec) {
+        throw boost::system::system_error(ec);
+      } else if (data[0] != Packet::MAGIC ||
+                 bytes_read < Packet::HEADER_LENGTH) {
+        return;
+      }
+      Packet(data, socket, k);
+    }
+  } catch (std::exception& e) {
+    std::cerr << "Exception in thread: " << e.what() << std::endl;
+  }
+}
+
+int main() {
+  auto k = std::make_shared<KeyValueStore>();
```

Comment:
```k -> key``` not that much different in terms of characters, but key is more clear, k sounds more like a looping variable.